rfd/0098-registered-openssh-nodes.md

```diff
@@ -0,0 +1,124 @@
+
+Pros:
+
+- No need to connect to node to perform RBAC check
+- Updating node labels can simply be done by using `tctl`
+- CA rotation is simple as every registered OpenSSH node will have the same CA
+
+Cons: none (that I can think of)
+
+#### My choice: Option 2
+
+Option 2 makes CA rotation and updating node labels much easier, and has the only downside of having to search for node resources before connecting to one. 
+
+### Security
+
+Both RBAC checks and session recording for registered OpenSSH nodes require that users connect through a Proxy. If users are able to connect to registered OpenSSH nodes directly, they can bypass both features. Currently the User CA public key is copied to OpenSSH nodes and configured to be trusted by `sshd`. This means OpenSSH nodes will accept certificates that are signed by the User CA. The problem with that is when Teleport users authenticate with a cluster, the Auth server replies with a certificate that is signed by the User CA. Users could potentially use this certificate to directly connect to registered OpenSSH nodes.
+
+Using the new Agentless CA to sign certificates used to authenticate with registered OpenSSH nodes solves this problem, as Teleport users do not have access to any certificates signed by the Agentless CA.
+
+### UX
+
+The following Teleport node features won't work with registered OpenSSH nodes:
+
+- Enhanced session recording and restricted networking
+- Host user provisioning
+- Session recording without SSH session termination
+- Dynamic labels
+- Outbound persistent tunnels to Proxies
+
+Due to this and other potential future differences, `tsh ls` and the node listing on the web UI should be updated to display if nodes are registered OpenSSH nodes or Teleport agent nodes.
+
+### Future work
+
+#### Session recording
+
+Currently session recording is required to be set be in `proxy` mode to work with OpenSSH nodes. That is not going to change, but ideally this requirement could be lifted when Teleport agent nodes and registered OpenSSH nodes are both in a single cluster. When establishing an SSH connection inside a cluster, depending on what the session recording mode is set to the appropriate type of session recording would be used:
+
+If session recording is in `node` or `node-sync` mode:
+
+- If the node is a Teleport Agent node, the node would record the session and upload it as normal.
+- If the node is a registered OpenSSH node, the Proxy would terminate and record the SSH session and upload it.
+
+If session recording is in `proxy` or `proxy-sync` mode:
+
+- Behavior would be unaffected. The Proxy would terminate, record and upload the session. This mode will still be required if users of a cluster wish to connect to unregistered OpenSSH nodes.
```

Comment:
What does it exactly mean, that the Proxy would "terminate and record the SSH session"?

Reply:
Just how Proxy recording mode works currently: https://goteleport.com/docs/architecture/session-recording/#record-at-proxy